### ISSUE TITLE
No Improvement (Model deterioration after training)

### ISSUE THREAD

**Issue by ZeeshanSultan** [mozilla/DeepSpeech#1574]
I have the latest version of the library and the model downloaded as of today.
I wanted to fine tune the model with my own dataset specifically automated voice samples 

The data consisted of 9 audio files:
transcripted as
>A1-converted.wav,please enter the mailbox number or press number sign to use the directory
>A2-converted.wav,please enter the mailbox number or press number sign to use the directory
>A3-converted.wav,please enter the mailbox number or press number sign to use the directory
>A4-converted.wav,please enter the mailbox number or press number sign to use the directory
>A5-converted.wav,please enter the mailbox number or press number sign to use the directory
>A6-converted.wav,please enter the mailbox number or press number sign to use the directory
>B1-converted.wav,please enter the mailbox number or press pound to use the directory
>B2-converted.wav,please enter the mailbox number or press pound to use the directory
>B3-converted.wav,please enter the mailbox number or press pound to use the directory

Here is the dataset file as well
[dataset.zip](https://github.com/mozilla/DeepSpeech/files/2408425/dataset.zip)

Initially before training the output transcription was as follows:
>'A1-converted.wav: please enter the mail box number or press number sign to use the directory'  
>'A2-converted.wav: leave entered the mail box a number of press number so in to use the directory'  
>'A3-converted.wav: he enters the mail box number or presnumbersinjunuwsetodirectory'  
>'A4-converted.wav: as entered the mail box number or press number sent us the directory'  
>'A5-converted.wav: please entered the mail box in number or press number sign to use the directory'  
>'A6-converted.wav: leave enter the mail box number or prestnumbersignedhepygerestory'  
>'B1-converted.wav: leave enter the mail box a number or press pound to use the directory '  
>'B2-converted.wav: leave enter the mail box number or press pound to use the directory'
>'B3-converted.wav: enter the mail box number or press pound to use the directory'  

Due to highly inaccurate results, I set the training to 10 epochs. for this i ran
`python3 DeepSpeech.py --n_hidden 2048 --initialize_from_frozen_model /model/output_graph.pb --checkpoint_dir fine_tuning_checkpoints --epoch 10 --train_files training_data.csv --dev_files training_data.csv --test_files training_data.csv --learning_rate 0.0001
`

And as a result the loss reached ~2.4 and the test epoch was almost 100% accurate:
<img width="1440" alt="screen shot 2018-09-23 at 9 00 59 am" src="https://user-images.githubusercontent.com/8667667/45924338-efe8d780-bf16-11e8-882d-376b97455486.png">

But supposedly the model only deteriorated after training because when i ran the transcription on the same data after training.
The results were as follows
>'A1-converted.wav: please enter the mail box number or press number sign to use the directory' (same)
>'A2-converted.wav: leave entered the mail box a number of press number so in to use the directory' (same)
>'A3-converted.wav: he enters the mail box number or presnumbersienjouwsete directory' (same)
>'A4-converted.wav: as entered the mail box number or press number sent us the directory' (same)
>'A5-converted.wav: please entered the mail box in number or press number sign to use the directory' (same)
>'A6-converted.wav: leave enter the mail box number or prestnumbersignedhepygerestory' (same)
>'B1-converted.wav: leave enter the mail box a number or press pound to use the directory' (same)
>'B2-converted.wav: leave enter the mail box number or press pound to use the directory' (same)
>'B3-converted.wav: ventured the mail box number or press pound to use the directory' **(deteriorated)**

What is the possible reason for this deterioration. and how to tackle this with the given dataset.

**Comment by kdavis-mozilla** [comment]
Thanks for trying out fine tuning with Deep Speech!

That said, we try to reserve issues for bugs or enhancements. General discussions on for example fine tuning are held in our [Discourse Forum](https://discourse.mozilla.org/c/deep-speech). So I'm wondering if you could move this discussion there where we can better address it. Thanks.
